util/src/main/java/io/kubernetes/client/util/authenticators/OpenIDConnectAuthenticator.java

```diff
@@ -0,0 +1,288 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.kubernetes.client.util.authenticators;
+
+import io.kubernetes.client.util.KubeConfig;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Scanner;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+import org.jose4j.json.internal.json_simple.JSONObject;
+import org.jose4j.json.internal.json_simple.parser.JSONParser;
+import org.jose4j.json.internal.json_simple.parser.ParseException;
+import org.jose4j.jws.JsonWebSignature;
+import org.jose4j.jwt.JwtClaims;
+import org.jose4j.jwt.MalformedClaimException;
+import org.jose4j.jwt.NumericDate;
+import org.jose4j.jwt.consumer.InvalidJwtException;
+import org.jose4j.lang.JoseException;
+
+/** OpenIDConnectAuthenticator */
+public class OpenIDConnectAuthenticator implements Authenticator {
+
+  /** The id_token */
+  public static final String OIDC_ID_TOKEN = "id-token";
+  /** The issuer */
+  public static final String OIDC_ISSUER = "idp-issuer-url";
+  /** The refresh_token */
+  public static final String OIDC_REFRESH_TOKEN = "refresh-token";
+  /** The client_id */
+  public static final String OIDC_CLIENT_ID = "client-id";
+  /** Optional client secret */
+  public static final String OIDC_CLIENT_SECRET = "client-secret";
+  /** Optional IdP TLS Certificate */
+  public static final String OIDC_IDP_CERT_DATA = "idp-certificate-authority-data";
+
+  static {
+    KubeConfig.registerAuthenticator(new OpenIDConnectAuthenticator());
+  }
+
+  @Override
+  public String getName() {
+    return "oidc";
+  }
+
+  @Override
+  public String getToken(Map<String, Object> config) {
+    return (String) config.get(OIDC_ID_TOKEN);
+  }
+
+  @Override
+  public boolean isExpired(Map<String, Object> config) {
+    String idToken = (String) config.get(OIDC_ID_TOKEN);
+
+    if (idToken == null) {
+      return true;
+    } else {
```

Comment:
Sorry I missed this. Please drop the `else` it's unnecessary given the `return` in the if clause.